jni_gen/test/config_provider_test.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:jni_gen/src/config/config_provider.dart';
+import 'package:test/test.dart';
+import 'package:path/path.dart' hide equals;
+
+void main() {
+  final config = ConfigProvider.parseArgs([
+    '--config',
+    join('test', 'test_data', 'test_config.yaml'),
+    '-Dsummarizer.use_asm=true',
+    '--override c_root=src/third_party',
+  ], allowYamlConfig: true);
+  test('getBool', () {
+    expect(config.getBool('summarizer.use_asm'), isTrue);
```

Comment:
If using a `Config` object all these test expectations _and_ all usages in the rest of the code become typed. (You could even implement the `operator==` on `Config`, and just check equality in the test here.)

`getString`, `getBool` etc would be private functions that populate this `Config` object.

Reply:
I decided to go with comparison method in test/ itself, since that will show what was expected and what was obtained.